spirv_reflect.h

```diff
@@ -440,11 +457,16 @@ typedef struct SpvReflectDescriptorBinding {
   SpvReflectBindingArrayTraits        array;
   uint32_t                            count;
   uint32_t                            accessed;
+  SpvReflectAccessFlags               access_flags;
   uint32_t                            uav_counter_id;
   struct SpvReflectDescriptorBinding* uav_counter_binding;
 
   SpvReflectTypeDescription*          type_description;
 
+  uint32_t                            is_byte_address_buffer;
```

Comment:
We should find some way to make this a flag so it can be used to identify other concrete types. ByteAddressBuffer is a concrete type on an SRV and there are a few others like it.

Reply:
sorry this is old (about to replace with new PR)

I created a 

```
typedef enum SpvReflectUserType {
  SPV_REFLECT_USER_TYPE_INVALID = 0,
  SPV_REFLECT_USER_TYPE_CBUFFER,
  SPV_REFLECT_USER_TYPE_TBUFFER,
  SPV_REFLECT_USER_TYPE_APPEND_STRUCTURED_BUFFER,
  ...
```

so this becomes an enum... basically will be a more general approach, this PR is **too** single focused to solve one task